src/pillowtalk_impl.c

```diff
@@ -630,13 +656,26 @@ static pt_response_t* http_operation(const char* http_method,
 
   curl_easy_setopt(curl_handle, CURLOPT_CONNECTTIMEOUT, 10);
 
+  /* specify header: Content-Type */
+  headers = curl_slist_append(headers, "Content-Type: application/json");
+  curl_easy_setopt(curl_handle, CURLOPT_HTTPHEADER, headers);
+
   // Want to avoid CURL SIGNALS
   curl_easy_setopt(curl_handle, CURLOPT_NOSIGNAL, 1);
 
   printf("%s : %s\n", http_method, server_target);
 
+  /* set the http verb */
   if (!strcmp("PUT", http_method))
+  {
     curl_easy_setopt(curl_handle, CURLOPT_UPLOAD, 1);
+    curl_slist_append(headers, "Transfer-Encoding: chunked");
+  }
+  else if (!strcmp("POST", http_method))
+  {
+    curl_easy_setopt(curl_handle, CURLOPT_POST, 1);
+    curl_slist_append(headers, "Transfer-Encoding: chunked");
+  }
```

Comment:
Is there a reason you used chunked instead of setting the length (which was noticeably missing, thanks)?  We have that information here.  I would certainly understand if we were using a callback function to stream data (which would be a nice improvement for the future).  

Anyways, just curious.